app/src/main/java/com/example/caregiver/EditTask.java

```diff
@@ -0,0 +1,167 @@
+        // navigate to dashboard after update
+        Intent intent = new Intent(this, Dashboard.class);
+        startActivity(intent);
+    }
+
+    /**
+     * Updates task in firebase, triggered when user clicks 'Update'
+     * If room is updated, the previous task is removed
+     * The updated task is written to the DB
+     * @param view
+     */
+    public void UpdateTask(View view) {
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
+        String path;
+        String updatedRoom = roomSpinner.getSelectedItem().toString();
+
+        // if room changed, remove previous task
+        if(!updatedRoom.equals(currTask.room)){
+            // if room is changed, remove the old task from its room
+            path = createPath(currTask.caregiveeId, currTask.room, currTask.taskId);
+            removeTaskInFirebase(path);
+        }
+
+        // post updated task details
+        String updatedTaskName = String.valueOf(taskNameField.getText());
+        String updatedNotes = String.valueOf(taskNotesField.getText());
+        // formulate path
+        path = createPath(currTask.caregiveeId, updatedRoom, currTask.taskId);
+        // create updated task HashMap
+        Map<String, Object> updatedTask = new HashMap<>();
+        updatedTask.put("assignedStatus", currTask.assignedStatus); // prev value
+        updatedTask.put("caregiverID", preferences.getString("userId", null)); // curr user's info
+        updatedTask.put("completionStatus", currTask.completionStatus); // prev value
+        updatedTask.put("name", updatedTaskName); // updated value
+        updatedTask.put("notes", updatedNotes); // updated value
+
+        // post updated task to path in Firebase
+        updateTaskInFirebase(path, updatedTask);
+    }
+
+    /**
+     * Creates path String for Firebase update/removal
+     * @param caregiveeId, ID of caregivee
+     * @param room, task room
+     * @param taskId, task ID
+     * @return string with path location
+     */
+    private String createPath(String caregiveeId, String room, String taskId){
+        return "/users/" + caregiveeId + "/rooms/" + room + "/tasks/" + taskId + "/";
+    }
+
+    /**
+     * In the case of a room change, this removes the task from the previous room
+     * @param path, path to task that needs to be removed
+     */
+    private void removeTaskInFirebase(String path){
+        DatabaseReference ref = database.child(path);
+        ref.removeValue();
+    }
+
+    /**
+     * Creates a ref to the inputted path to task id in the DB & updates task details
+     * @param path, path to taskId
+     * @param updatedTask, HashMap with task details (assignedStatus, CaregiverID, CompletionStatus,
+     *                     Name, Notes
+     */
+    private void updateTaskInFirebase(String path, Map<String, Object> updatedTask){
+        DatabaseReference ref = database.child(path);
+        ref.updateChildren(updatedTask);
```

Comment:
We should add success and error message when user update task instead of redirecting. In case an update task is not successful. So something like: 

```java
ref.updateChildren(updatedTask, new DatabaseReference.CompletionListener() {@Override
   public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
          if (databaseError == null) {
              displaySuccessMessage("Your task is updated"); 
          } else {
              displayErrorMessage("Your task cannot be updated"); 
          }
      }
 });
 ```
 
 The display success message and error message you can copy over the function `displayMessage` from my AddTask class. Feel free to put this issue over with the 'delete' PR if it's too much work to do the error handling here.

Reply:
Yes! thank you for pointing that out